### DISCUSSION TITLE
Cannot set isIndeterminate=true for CircularProgress as default prop in theme extension

### DISCUSSION THREAD

**Discussion by jonamat** [chakra-ui/chakra-ui#5898] [Q&A]
I want to avoid providing the prop "isIndeterminate" to every instance of CircularProgress. I followed the docs to https://chakra-ui.com/docs/styled-system/theming/customize-theme but for some components it seems to not work.

I can't see CircularProgress in this page https://github.com/chakra-ui/chakra-ui/blob/main/packages/theme/src/components/progress.ts

It seems that some components are built on top of other components and I struggle every time I try to customize them.

Can you explain me how actually extending every ChakraUI component? (not all obviusly, but one or more generic methods to extend them)

```tsx
export const defaultTheme = extendTheme((theme: Theme) =>

  mergeThemeOverride(theme, {
    components: {
      CircularProgress: {
        defaultProps: {
          isIndeterminate: true,
        },
      },
    },
  }),
);
```

**Comment by chr-ge** [comment]
You can only use the theme for styling and therefore you cannot use non-style props in the theme. If you want to have `isIndeterminate={true}` on every `CircularProgress`, you need to create a custom `CircularProgress` component.
